ssl/quic/quic_rx_depack.c

```diff
@@ -0,0 +1,612 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <internal/packet.h>
+#include <internal/nelem.h>
+#include <internal/quic_wire.h>
+#include <internal/quic_record_rx.h>
+#include <internal/quic_ackm.h>
+
+#include "quic_local.h"
+#include "../ssl_local.h"
+
+/*
+ * ASSUMPTION: the QUIC_CONNECTION structure refers to other related
+ * components, such as OSSL_ACKM and OSSL_QRX, in some manner.  These macros
+ * should be used to get those components.
+ */
+#if 0                            /* Currently unimplemented */
+# define GET_CONN_ACKM(c) ((c)->ackm)
+#else
+# define GET_CONN_ACKM(c) NULL
+#endif
+#if 0                            /* Currently unimplemented */
+# define GET_CONN_QRX(c) ((c)->qrx)
+#else
+# define GET_CONN_QRX(c) NULL
+#endif
+#if 0                            /* Currently unimplemented */
+# define GET_CONN_STATM(c) ((c)->statm)
+#else
+# define GET_CONN_STATM(c) NULL
+#endif
+
+/* =====BEGIN TO BE REMOVED===== */
```

Comment:
BTW, I suggest we adopt a standard mechanism for noting TODO items (like we have for 3.0, and other similar things in the past), e.g.

```` C
/* TODO(QUIC): whatever */
````

Reply:
The reason I made this marker is to wrap a whole section that should end up removed, and to make it clear where it begins and ends.  But perhaps something like this?

``` C
/* TODO(QUIC): [BEGIN: TO BE REMOVED] placeholder macros and functions */
...
/* TODO(QUIC): [END: TO BE REMOVED] */

Reply:
Done